compiler/rustc_hir_analysis/src/astconv/lint.rs

```diff
@@ -241,13 +241,11 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
             } else {
                 let msg = "trait objects without an explicit `dyn` are deprecated";
                 tcx.node_span_lint(BARE_TRAIT_OBJECTS, self_ty.hir_id, self_ty.span, msg, |lint| {
-                    if self_ty.span.can_be_used_for_suggestions()
-                        && !self.maybe_lint_impl_trait(self_ty, lint)
-                    {
+                    if self_ty.span.can_be_used_for_suggestions() {
```

Comment:
Now we should be able to remove the `diag.is_error()` inside `maybe_lint_impl_trait` again simplifying the latter.

Reply:
I didn't bother removing them earlier because it didn't affect the line count at all (we already have to check for `is_downgradable` anyways. Regardless, done.